R/update-version.R

```diff
@@ -55,3 +55,43 @@ get_date <- function() {
   author_time <- parsedate::parse_iso_8601(Sys.getenv("GIT_COMMITTER_DATE"))
   as.Date(author_time)
 }
+
+fledge_bump_version <- function(desc, which) {
+  if (which %in% c("patch", "minor", "major", "dev")) {
+    desc$bump_version(which)
```

Comment:
Should we forbid > 99 here? We can roll our own bumping, we won't need `suppressMessages()` above too.

Perhaps:

```r
new_version <- switch(which,
  "dev" = c(major, minor, patch, (dev + 1) %||% 9000),
  "patch" = {
    if (patch >= 99) abort(...)
    c(...)
  },
  "minor" = ...
)
```

with explicit local variables for the four components?

Reply:
> Should we forbid > 99 here

And abort if patch/minor are already higher? I had wondered about this in a comment too.

> We can roll our own bumping, we won't need suppressMessages() above too.

- Actually the desc issue was solved so that's not needed anyway.
- I find it easier to separate both cases as we need to check the fourth component (dev) is present for "pre-minor" and "pre-major" (at least I assume so: one can't go from _not_ dev to "pre-minor"/"pre-major"?)

Reply:
We want to abort if `patch >= 99`, as in the code example, overall it looks like it will be simpler without `desc::bump_version()` .

Reply:
I added the error. Does the behavior in the test file look ok?

Why would removing

```r
  if (which %in% c("patch", "minor", "major", "dev")) {
    desc$bump_version(which)
    return(desc)
  }

```

be simpler?

Reply:
I feel the current code gets the "easiest cases" out of the way right away but I can amend further obviously.